lxdock/guests/base.py

```diff
@@ -175,6 +182,10 @@ def copy_directory(self, host_path, guest_path):
         self.run(['tar', '-xf', guest_tar_path, '-C', str(guest_path)])
         self.run(['rm', '-f', str(guest_tar_path)])
 
+    def is_provisioned(self):
+        """ Returns a boolean indicating if the container is provisioned. """
+        return self.lxd_container.config.get('user.lxdock.provisioned') == 'true'
+
```

Comment:
The proper conceptual place for `is_provisioned()` is fuzzy. `is_provisioned()` is tightly bound to `setup()` because the flag is there partly because we want to ensure that `setup()` is only called once.

The main reason why `is_provisioned()` moved in `Guest` is that it happened *before* I changed `lxd_container` to `container` in the initializer. So at the time, the move was necessary. But then, now that it's not necessary any more, maybe it would be a good idea to move it back in `Container`, yes.

As for your "how about", it's already what we do with the `guests = [g for g in guests if not g.is_provisioned()]`. I don't understand your question.